src/converse-sendbutton.js

```diff
@@ -0,0 +1,94 @@
+// Converse.js (A browser based XMPP chat client)
+// http://conversejs.org
+//
+// Copyright (c) 2012-2017, Jan-Carel Brand <jc@opkode.com>
+// Licensed under the Mozilla Public License (MPLv2)
+//
+/*global Backbone */
+
+/* This plugin is to add an option of to have send button in chat box */
+
+(function(root, factory) {
+    define([
+        "converse-core",
+        "converse-chatview",
+        "converse-muc",
+        "converse-headline"
+    ], factory);
+}(this, function(converse) {
+    "use strict";
+    var $ = converse.env.jQuery;
+
+    converse.plugins.add('converse-sendbutton', {
+
+        overrides: {
+            // Overrides mentioned here will be picked up by converse.js's
+            // plugin architecture they will replace existing methods on the
+            // relevant objects or classes.
+            //
+            // New functions which don't exist yet can also be added.
+
+            ChatBoxView: {
+                events: {
+                    'click .send-button': 'onSendButtonClicked'
+                },
+
+                onSendButtonClicked: function(ev) {
+                    /* Event handler for when a send button is clicked in a chat box textarea.
+                     */
+                    var _converse = this.__super__._converse;
+                    ev.preventDefault();
+                    var textarea = $(ev.target).siblings('.chat-textarea')[0],
+                        message;
+                    message = textarea.value;
+                    textarea.value = '';
+                    textarea.focus();
+                    if (message !== '') {
+                        this.onMessageSubmitted(message);
+                        _converse.emit('messageSend', message);
+                    }
+                    this.setChatState(_converse.ACTIVE);
+                },
+
+                render: function() {
+                    /* Render the UI container in which chat room messages will
+                     * appear.
+                     */
+                    this.__super__.render.apply(this, arguments);
+                    this.$('.sendXMPPMessage').append('<button type="button" class="pure-button send-button">Send</button>');
```

Comment:
That would go against the whole idea of making this a plugin. By making it a plugin we want to keep it separate from other code.

However, I'm now starting to doubt whether it was a good idea of mine to suggest that we make it a plugin. Perhaps it's better after all to have all this in `converse-chatview.js`.

What originally made me think it would be better as a plugin was because of the fact that you changed the CSS in such a way that it would break without the button. However we can solve that by simply adding a new class to the textarea if the button must be shown and then adding the special button-specific CSS to that class.

So actually, I think it would be best to keep the code in `converse-chatview.js` where you had it originally. I'm sorry for flip-flopping and having given you extra seemingly unnecessary work.

However, the silver lining is that it was good practice. If you're going to work on a GSoC project, then you will definitely have to create a new plugin, so the knowledge you gained now wasn't for nothing.